### PR TITLE
feat(l2): prepare contracts for withdraw

### PR DESCRIPTION
**Pull request by ilitteri**
**Motivation**

To complete the withdrawal flow

**Description**

- [Add openzeppelin-contracts dep installation](https://github.com/lambdaclass/lambda_ethereum_rust/commit/a23066b46712ac109cbbe27c1c696dd529b331c9)
- [Implement withdrawals in CommonBridge](https://github.com/lambdaclass/lambda_ethereum_rust/commit/165d1e84fe58fc8803120454cf36f48800e872d0)
- [Update ICommonBridge](https://github.com/lambdaclass/lambda_ethereum_rust/commit/57c3aca9af22e85afd05ce716e7fa0fd1c049626)
- [Implement commit method in OnChainProposer](https://github.com/lambdaclass/lambda_ethereum_rust/commit/190122089b03aa8b74e5548130d81416f716935d)
- [Update IOnChainProposer with new commit def](https://github.com/lambdaclass/lambda_ethereum_rust/commit/58010b9153837dc96a7cbc1173de9f1ee47862bb)
- [Update deploy script](https://github.com/lambdaclass/lambda_ethereum_rust/commit/73d2b8f609495863db18403804333595deec1f18)

### CODE REVIEW

File: crates/l2/contracts/src/l1/CommonBridge.sol [hunk 1]
```diff
@@ -8,14 +8,44 @@ import "./interfaces/ICommonBridge.sol";
...
 /// @author LambdaClass
 contract CommonBridge is ICommonBridge, Ownable, ReentrancyGuard {
-    constructor(address owner) Ownable(owner) {}
+    /// @notice Mapping of unclaimed withdrawals. A withdrawal is unclaimed if
+    /// there is a non-zero value in the mapping (a merkle root) for the hash
+    /// of the L2 transaction that requested the withdrawal.
+    /// @dev The key is the hash of the L2 transaction that requested the
+    /// withdrawal.
+    /// @dev The value is a boolean indicating if the withdrawal was claimed or not.
+    mapping(bytes32 => bool) public claimedWithdrawals;
```
Comment (ManuelBilbao) [inline crates/l2/contracts/src/l1/CommonBridge.sol:17]: Maybe this should be `unclaimedWithdrawals`
```suggestion
    mapping(bytes32 => bool) public unclaimedWithdrawals;
```

File: crates/l2/contracts/src/l1/CommonBridge.sol [hunk 2]
```diff
@@ -8,14 +8,44 @@ import "./interfaces/ICommonBridge.sol";
...
+            msg.sender == ON_CHAIN_PROPOSER,
+            "CommonBridge: caller is not the OnChainProposer"
+        );
+        _;
     }
 
-    mapping(bytes32 l2TxHash => WithdrawalData) public pendingWithdrawals;
+    constructor(address owner) Ownable(owner) {}
+
+    function initialize(address onChainProposer) public nonReentrant {
```
Comment (ManuelBilbao) [inline crates/l2/contracts/src/l1/CommonBridge.sol:38]: Should we limit this function to be called only once?
Maybe adding a `require(ON_CHAIN_PROPOSER == 0)` so once it's changed it cannot be changed anymore

File: crates/l2/contracts/src/l1/OnChainProposer.sol [hunk 3]
```diff
@@ -1,14 +1,68 @@
...
+    /// @dev If a block is verified, the block hash is stored here.
+    /// @dev If a block was not verified yet, it won't be here.
+    /// @dev It is used by other contracts to verify if a block was verified.
+    mapping(uint256 => bool) public verifiedBlocks;
+
+    address public BRIDGE;
+
+    constructor() {}
+
+    function initialize(address bridge) public nonReentrant {
```
Comment (ManuelBilbao) [inline crates/l2/contracts/src/l1/OnChainProposer.sol:29]: The same here about only call this method once

File: crates/l2/contracts/src/l1/CommonBridge.sol (hunk 2 above)
Comment (ilitteri) [reply crates/l2/contracts/src/l1/CommonBridge.sol:38]: Yes! Makes sense.

File: crates/l2/contracts/src/l1/CommonBridge.sol (hunk 1 above)
Comment (ManuelBilbao) [reply crates/l2/contracts/src/l1/CommonBridge.sol:17]: Then I think we should change the docs:
```
/// @notice Mapping of unclaimed withdrawals. A withdrawal is unclaimed if
/// there is a non-zero value in the mapping (a merkle root) for the hash
/// of the L2 transaction that requested the withdrawal.
```
It should be "A withdrawal is claimed if there is a non-zero value..."

Comment (ilitteri) [reply crates/l2/contracts/src/l1/CommonBridge.sol:17]: Done [191eb47](https://github.com/lambdaclass/lambda_ethereum_rust/pull/944/commits/191eb47b78ab4dba1ee4ff478750d5428afee2ad)

File: crates/l2/contracts/src/l1/CommonBridge.sol (hunk 2 above)
Comment (ilitteri) [reply crates/l2/contracts/src/l1/CommonBridge.sol:38]: Done [285f41c](https://github.com/lambdaclass/lambda_ethereum_rust/pull/944/commits/285f41c8326b1899788ad0b8313477741f5313ca)

File: crates/l2/contracts/src/l1/OnChainProposer.sol (hunk 3 above)
Comment (ilitteri) [reply crates/l2/contracts/src/l1/OnChainProposer.sol:29]: Done [285f41c](https://github.com/lambdaclass/lambda_ethereum_rust/pull/944/commits/285f41c8326b1899788ad0b8313477741f5313ca)

**Pull request merged**